packages/common/src/utils/generateReports/generateRootSummaryDataCSV.ts

```diff
@@ -0,0 +1,135 @@
+  let marketingCookies = 0;
+  let uncategorisedCookies = 0;
+  let cookiesWithIssues = 0;
+  let analyticsCookiesWithIssues = 0;
+  let functionalCookiesWithIssues = 0;
+  let marketingCookiesWithIssues = 0;
+  let uncategorisedCookiesWithIssues = 0;
+  let totalCookies = 0;
+
+  // eslint-disable-next-line complexity
+  Object.keys(extractedData).forEach((cookieKey) => {
+    const cookie = extractedData[cookieKey];
+
+    if (!cookie.analytics) {
+      return;
+    }
+    if (cookie.isFirstParty) {
+      totalFirstPartyCookies += 1;
+    } else {
+      totalThirdPartyCookies += 1;
+    }
+
+    if (
+      cookie.isBlocked ||
+      (cookie.blockedReasons && cookie.blockedReasons?.length > 0)
+    ) {
+      cookiesWithIssues += 1;
+    }
+
+    switch (cookie.analytics.category) {
+      case 'Analytics':
+        analyticsCookies += 1;
+        if (
+          cookie.isBlocked ||
+          (cookie.blockedReasons && cookie.blockedReasons?.length > 0)
+        ) {
+          analyticsCookiesWithIssues += 1;
+        }
+        break;
+      case 'Functional':
+        functionalCookies += 1;
+        if (
+          cookie.isBlocked ||
+          (cookie.blockedReasons && cookie.blockedReasons?.length > 0)
+        ) {
+          functionalCookiesWithIssues += 1;
+        }
+        break;
+      case 'Marketing':
+        marketingCookies += 1;
+        if (
+          cookie.isBlocked ||
+          (cookie.blockedReasons && cookie.blockedReasons?.length > 0)
+        ) {
+          marketingCookiesWithIssues += 1;
+        }
+        break;
+      case 'Uncategorized':
+        uncategorisedCookies += 1;
+        if (
+          cookie.isBlocked ||
+          (cookie.blockedReasons && cookie.blockedReasons?.length > 0)
+        ) {
+          uncategorisedCookiesWithIssues += 1;
+        }
+        break;
+      default:
+        break;
+    }
+    totalCookies += 1;
+  });
+
+  const summary = {
+    [I18n.getMessage('totalCookies')]: totalCookies,
+    [I18n.getMessage('totalFirstPartyCookies')]: totalFirstPartyCookies,
+    [I18n.getMessage('totalThirdPartyCookies')]: totalThirdPartyCookies,
+    [I18n.getMessage('analyticsCookies')]: analyticsCookies,
+    [I18n.getMessage('functionalCookies')]: functionalCookies,
+    [I18n.getMessage('marketingCookies')]: marketingCookies,
+    [I18n.getMessage('uncategorizedCookies')]: uncategorisedCookies,
+    [I18n.getMessage('cookiesWithIssues')]: cookiesWithIssues,
+    [I18n.getMessage('analyticsCookiesWithIssues')]: analyticsCookiesWithIssues,
+    [I18n.getMessage('functionalCookiesWithIssues')]:
+      functionalCookiesWithIssues,
+    [I18n.getMessage('marketingCookiesWithIssues')]: marketingCookiesWithIssues,
+    [I18n.getMessage('uncategorizedCookiesWithIssues')]:
+      uncategorisedCookiesWithIssues,
+  };
+
+  const CSVString = Object.entries(summary).reduce(
+    (acc, [key, value]) => (acc += `${key}, ${value}\r\n`),
+    ''
+  );
+
+  return CSVString;
```

Comment:
```suggestion
 return Object.entries(summary)
    .map(([key, value]) => `${key}, ${value}`)
    .join('\r\n');
```

Will that return the same result?

Reply:
`.join` would not add `\r\n` at the end of the string. We can modify and add it in the string before returning.